whylog/teacher/__init__.py

```diff
@@ -1,49 +1,78 @@
 from whylog.teacher.mock_outputs import create_sample_rule
-from whylog.teacher.user_intent import UserRuleIntent
+
+
+class TeacherParser(object):
+    def __init__(self, line_object):
+        self.line = line_object
+        self.pattern_name = None
+        self.log_type = None
+        self.primary_keys = []
+
+
+class TeacherConstraint(object):
+    def __init__(self):
+        self.constr_type = None
+        self.params = {}
 
 
 class Teacher(object):
     """
     Enable teaching new rule. One Teacher per one entering rule.
     """
 
-    def __init__(self, id_to_line_dict, effect_id, config, pattern_assistant):
-        self._lines = id_to_line_dict.copy()
-        self.rule_intent = UserRuleIntent(effect_id)
+    def __init__(self, config, pattern_assistant):
+        self._constraints = {}  # dict of TeacherConstraint
```

Comment:
dict maps from something to something. I assume value is `TeacherConstraint` object, but what is the dictionary key?